util/topgen/c.py

```diff
@@ -165,19 +164,18 @@ def memories(self):
             ret.append((m["name"],
                         MemoryRegion(self._top_name +
                                      Name.from_snake_case(m["name"]),
-                                     int(m["base_addr"][self.addr_space], 0),
-                                     int(m["size"], 0))))
+                                     int(m["base_addr"][m["addr_space"]], 0),
+                                     int(m["size"], 0))),
+                        m["addr_space"])
```

Comment:
Because TopGenC should be generating for a specific addr space, this could just be...

```suggestion
                        self.addr_space)
```

and we should keep that as a member field.